### PR TITLE
Ah pi gen

### PR DESCRIPTION
**Pull request by jvonau**
rebase of py-gen branch to just what is need to complete the image support

### CODE REVIEW

File: roles/0-init/tasks/hostname.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 - name: 'Turn the crank for systemd: hostnamectl set-hostname "{{ iiab_hostname }}.{{ iiab_domain }}" (debuntu)'
   shell: hostnamectl set-hostname "{{ iiab_hostname }}.{{ iiab_domain }}"
-  when: is_debuntu | bool
+  when: is_debuntu | bool and not installing
```
Comment (holta) [inline roles/0-init/tasks/hostname.yml]:
```suggestion
  when: is_debuntu and not installing
```

File: roles/4-server-options/tasks/main.yml [hunk 2]
```diff
@@ -5,10 +5,10 @@
     path: "{{ iiab_state_file }}"
     state: touch
 
-- name: Install dnsmasq
-  include_tasks: roles/network/tasks/dnsmasq.yml
-  when: dnsmasq_install | bool
-  tags: base, domain, dnsmasq, network
+- name: MYSQL setup
```
Comment (holta) [inline roles/4-server-options/tasks/main.yml]:
```suggestion
- name: MySQL setup
```

File: roles/db-setup/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+- name: Provision MySql
```
Comment (holta) [inline roles/db-setup/tasks/main.yml]:
```suggestion
- name: Provision MySQL
```

File: roles/mysql/tasks/main.yml [hunk 4]
```diff
@@ -115,84 +115,20 @@
...
-- name: Update MySQL root password for localhost root accounts, if mysql_enabled
-  mysql_user:
-    name: root
-    host: localhost
-    password: "{{ mysql_root_password }}"
-    priv: "*.*:ALL,GRANT"
-  when: mysql_enabled | bool
-
-- name: Install .my.cnf file from template, with root password credentials, if mysql_enabled
+- name: Use custom systemd unit file to call runrole db-setup on boot
```
Comment (holta) [inline roles/mysql/tasks/main.yml]:
```suggestion
- name: Use custom systemd unit file (iiab-setup-db.service) to call runrole db-setup on boot
```

File: roles/mysql/tasks/main.yml [hunk 5]
```diff
@@ -115,84 +115,20 @@
...
-      value: '"MySQL is a widely used free and open source (GPLv2) database, offered by most web hosting services, on a diversity of platforms."'
-    - option: enabled
-      value: "{{ mysql_enabled }}"
+    name: iiab-setup-db.service
+    enabled: yes
+  when: installing | bool
+
+- name: Provison the DB
+  include_tasks: setup.yml
+  when: not installing | bool
```
Comment (holta) [inline roles/mysql/tasks/main.yml]:
```suggestion
  when: not installing
```

File: roles/wordpress/tasks/main.yml [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
 
-- name: Provision MySql DB for WordPress
-  include_tasks: setup.yml
-  when: wordpress_install | bool and not installing | bool
-
 - name: Install WordPress if wordpress_installed is absent
   include_tasks: install.yml
   when: wordpress_install and not wordpress_installed is defined
 
+- name: Provision MySql DB for WordPress
```
Comment (holta) [inline roles/wordpress/tasks/main.yml]:
```suggestion
- name: Provision MySQL DB for WordPress
```

File: roles/4-server-options/tasks/main.yml (hunk 2 above)
Comment (jvonau) [reply roles/4-server-options/tasks/main.yml]: sure your highness

**Pull request closed without merging**